modules/src/ics03_connection/error.rs

```diff
@@ -10,6 +10,21 @@ pub enum Kind {
     #[error("connection state unknown")]
     UnknownState,
 
+    #[error("connection exists (was initialized) already")]
+    ConnectionExistsAlready,
```

Comment:
It would be useful to store some more metadata (like the identifier of the connection) in the errors themselves:

```suggestion
    #[error("connection exists (was initialized) already: {0}")]
    ConnectionExistsAlready(ClientId),
```

Reply:
Whenever I use `Kind::` syntax, I usually pass a context. For example:

```rust
Err(Kind::InvalidProof.context(String::from("consensus state proof is invalid"))
                        .into())
```

The alternative, based on your suggestion, would be to have:

```rust
Err(Kind::InvalidProof(String::from("consensus state proof is invalid"))
                        .into())
```
Would these be these the same thing? If so, not clear to me which is preferable or idiomatic.

Reply:
The purpose of `.context` is to provide additional context as to why the error was raised at this specific place / in these specific circumstances, not to provide an error message or to attach metadata to the error. The error message is defined by the `Display` impl automatically derived by the `thiserror` crate via it's `#[display]` annotation, while metadata that is specific to the error itself, as opposed to the circumstances of where and why it was raised, should be embedded within the error itself, much like the `ClientId` above.

Does that make thing clearer?